src/file.cr

```diff
@@ -72,6 +102,20 @@ class File < IO::FileDescriptor
     Crystal::System::File.exists?(path)
   end
 
+  # Returns the size of *filename* bytes. Raises `Errno` if the file at *path*
+  # does not exist.
+  #
+  # ```
+  # File.size("foo") # raises Errno
+  # File.write("foo", "foo")
+  # File.size("foo") # => 3
+  # ```
+  def self.size(filename) : UInt64
+    stat(filename).size
+  rescue ex : Errno
+    raise Errno.new("Error determining size of #{filename.inspect}", ex.errno)
```

Comment:
Can we avoid the raise/rescue/raise again chain?

Reply:
You can't avoid it entirely but I think it's possible in the case of `ENOENT` or `ENOTDIR`.

Reply:
Since we assume in `stat` that we can raise en exception and have `Errno.value` set correctly from the `stat(2)` syscall, I believe we can assume it here, too?

```crystal
stat?(filename).try(&.size) || raise Errno.new("stat")
```

Reply:
@ysbaddaden my point was that `stat?` can still raise, so it's not a complete solution.

Reply:
> `stat?` can still raise.

What?

Reply:
@ysbaddaden `stat?` still raises on EACCES or ELOOP or ENOMEM etc.